### DISCUSSION TITLE
How to read INT32 with LSW first?

### DISCUSSION THREAD

**Discussion by sauvant** [eModbus/eModbus#220] [Q&A] [answered]
Hi all,

I am trying to read some INT32 registers from an energy meter (https://gavazzi.se/app/uploads/2020/11/em330_em340_et330_et340_cp.pdf). The data format used is a follows:

"For all the formats the byte order (inside the single word) is MSB->LSB. In INT32, UINT32 and UINT64
formats, the word order is LSW-> MSW."

getting these values like this:
uint32_t w_l1;
msg.get(3, w_l1);
gives wrong values as it expects the most significant word first.

Any idea how to handle that elegantly?

Thanks,
Keith

**Comment by Miq1** [comment, marked as answer]
Um, that is a unfortunate interpretation of the Modbus standards... I am afraid there is no better way than reversing the words yourself.
